### DISCUSSION TITLE
How do I add a comments section to a Quarto blog?

### DISCUSSION THREAD

**Discussion by joshuaquinlan** [quarto-dev/quarto-cli#8581] [Q&A]
### Description

hi all. i am looking to add a comments section to my quarto blog but i haven't found a solution for this that fits my particular needs. the most popular solutions appear to be [utterances](https://utteranc.es/) and [giscus](https://giscus.app/), but they require users to sign in through github. i would prefer a solution that allowed people to comment anonymously or without a separate login, if possible. this would be more like the website "guestbooks" of yore, i suppose, or like the 4chan style of anonymous commenting. does anyone have any recommendations for how i might implement something like this? i toyed around with a couple of options a couple months ago (lost now, unfortunately, as this was in november and my memory is not quite that strong) but they didn't seem like they were compatible with quarto—at least from my extremely novice perspective.
thanks in advance!

edit: i remembered one of the options that i toyed with: [remark42](https://remark42.com/docs/getting-started/installation/). i struggled getting it to work with quarto though and am ultimately unsure as to whether it can work.

**Comment by mcanouil** [comment]
Quarto "builtin" commenting features are listed in <https://quarto.org/docs/output-formats/html-basics.html#commenting>.

If you want something else then you have to figure out what system that is and it would need to work for static website.
Now, assuming your find the tool you want, it's very likely HTML/JavaScript. This can be integrated in your website using:
- `include-`: <https://quarto.org/docs/output-formats/html-basics.html#includes>
- raw blocks: ` ```{=html} ` or inline `` `whatever HTML`{=html} ``

Related to:
- https://github.com/quarto-dev/quarto-cli/discussions/2168

**Reply by joshuaquinlan** [reply]
ah, thank you! yes, i suppose what i am looking for is a recommendation for a system of that nature, given that the "built-in" commenting features don't really suit my needs.

**Reply by mcanouil** [reply]
Well Quarto is not really the best place to ask for HTML/JavaScript technologies.
You'll have better luck elsewhere on GitHub or other platforms dedicated to these technologies.

This being said, it's possible that someone from the community that did implement something come across this post.

**Reply by harrelfe** [reply]
Even though `hypothes.is` requires an account, the functionality makes it worth using.  I really like how users can post either public of private (viewable only to the reader) comments, questions, answers, highlighting.

**Reply by edenian-prince** [reply]
@joshuaquinlan what did you end up going with?

I found some blogs that have commenting set up through github. https://cameronpatrick.com/post/2023/07/quarto-thesis-formatting/

I haven't explored them myself, but looks like quarto has other commenting tools built-in, not just hypothesis https://quarto.org/docs/reference/projects/websites.html#comments

**Reply by joshuaquinlan** [reply]
> @joshuaquinlan what did you end up going with?
> 
> I found some blogs that have commenting set up through github. https://cameronpatrick.com/post/2023/07/quarto-thesis-formatting/
> 
> I haven't explored them myself, but looks like quarto has other commenting tools built-in, not just hypothesis https://quarto.org/docs/reference/projects/websites.html#comments

i ended up just using utteranc.es due to its ease of implementation, but unfortunately it's quite useless for my purposes. i wanted to implement anonymous commenting so that people could easily just leave a thought/question/whatever. requiring the reader to sign into github is enough of an obstacle to stifle most commenting, i think (especially since many people don't have even have github accounts). i'd still love a method that allowed anonymous commenting!

**Reply by JCfly3000** [reply]
I am also looking for quarto Anonymous comment tool. Anyone have found one ?

**Reply by harrelfe** [reply]
I still go with hypothes.is - it's as anonymous as the hypothes.is account creator's profile wants it to be, and users can post comments that only they can see, optionally.

**Comment by Angelinarose1** [comment]
Beyond recognizes the critical role that environmental protection plays in sustainable business practices. We are determined to reduce our ecological footprint by proactively protecting the environment while meeting our industrial needs. This engagement extends throughout our supply chain, from extraction to final delivery, guaranteeing practices that respect the earth and meet the most strict environmental standards.

<a href="https://beyondgroup.ae/">Best Building Material Supplier</a>

**Comment by davidleitch1** [comment]
Just recently I set up Isso comments for my articles. You start an Isso server and then I wrote an html page that loads the java script. Then in those articles I want to include comments in I include the comments file with an include after body
```title: "Code testing"
author: "David Leitch"
date: "2024-08-25"
categories: [analysis,nem,wind]
image: "../media/image-20240828205204106.png"
lightbox: true
format:
  html:
    include-after-body:
       - "../comment_load.html"
  docx: default
draft: false
```

the html page:
```
<script data-isso="//myserver/"
        data-isso-css="true"
        data-isso-css-url="null"
        data-isso-lang="en"
        data-isso-max-comments-top="10"
        data-isso-max-comments-nested="5"
        data-isso-reveal-on-click="5"
        data-isso-avatar="false"
        data-isso-avatar-bg="#f0f0f0"
        data-isso-avatar-fg="#9abf88 #5698c4 #e279a3 #9163b6 ..."
        data-isso-vote="true"
        data-isso-vote-levels=""
        data-isso-page-author-hashes="f124cf6b2f01,7831fe17a8cd"
        data-isso-reply-notifications-default-enabled="false"
        src="https://myserver/js/embed.min.js"></script>
        <section id="isso-thread" data-title="Test tag" ></section>
```
I am only an aged amateur (finance is my thing, now electricity and decarbonisation) so it took a little while to get it running, things like trailing slashes can make a difference. However like a lot of programming stuff its rewarding when it eventually works. The server runs on my own server box via a cloudflare tunnel, the comments end up on my website which runs on netlify. A little knowledge is indeed a dangerous thing. Quarto is an excellent publishing system for a single person, together with Holoviz Panel you can move to a web based life. Publishing can be improved to be more incremental.

**Reply by mcanouil** [reply]
You could embed all this as a Quarto extension, see <https://quarto.org/docs/extensions/>.

**Reply by davidleitch1** [reply]
Perhaps after a bit more testing. Although my coding leaves much to be desired.

**Comment by travelmagnet462** [comment]
Using a Disqus Integration:

One of the easiest ways to add a comments section is by integrating Disqus, a popular comment system. You can embed it into your Quarto blog by modifying the layout file and adding the Disqus embed code into your blog's HTML. You'll need to sign up for Disqus and configure the settings on their platform first.
Using GitHub Issues:

Another approach is to leverage GitHub Issues for comments, especially if you’re hosting your Quarto site on GitHub Pages. This method allows users to open an issue for discussion on each blog post. You can add the necessary links or a button to enable users to open an issue directly from your blog.
Custom Solutions:

You could also consider using other third-party comment platforms, such as Commento or Staticman. Both offer simple integrations with static websites like those created with Quarto.
Adding a comments section not only helps engage your readers but also builds a community around your content. For example, tools like [PenetoLabs](https://penetolabs.com/) could be beneficial for content creators who want to optimize their decision-making based on user feedback. Using insights from comment interactions could enhance your content strategy and improve user engagement with your posts.

**Comment by mcanouil** [comment]
This post is against the CoC as it's clearly from an AI.
Additionally, it brings no real information.

The documentation about commenting: https://quarto.org/docs/output-formats/html-basics.html#commenting

> [!IMPORTANT]
> We appreciate your interest in the Quarto project, but the content above violates our [Code of Conduct](https://github.com/quarto-dev/quarto-cli?tab=coc-ov-file#readme).
>
> As a member of the Quarto community, you are expected to follow our Code of Conduct, which outlines the standards of behavior and communication that we value. The Code of Conduct applies to all online interactions within the Quarto project, and it is enforced by the Quarto team. By participating in the Quarto community, you agree to abide by the Code of Conduct and respect the rights and dignity of others. You can find the Code of Conduct at <https://github.com/quarto-dev/quarto-cli?tab=coc-ov-file#readme>.
> 
> Thank you for your cooperation and contribution. 🙏
